interfaces/builtin/serial_port.go

```diff
@@ -43,7 +43,13 @@ func (iface *SerialPortInterface) String() string {
 
 // Pattern to match allowed serial device nodes, path attributes will be
 // compared to this for validity when not using udev identification
-var serialDeviceNodePattern = regexp.MustCompile("^/dev/tty[A-Z]{1,3}[0-9]{1,3}$")
+// Known device node patterns we need to support
+//  - ttyUSBX  (UART over USB devices)
+//  - ttyACMX  (ACM modem devices )
+//  - ttyXRUSBx  (Exar Corp. USB UART devices)
+//  - ttySX (UART serial ports)
+//  - ttyOX (UART serial ports on ARM)
+var serialDeviceNodePattern = regexp.MustCompile("^/dev/tty[A-Z]+[0-9]+$")
```

Comment:
@niemeyer - I'm completely fine with being more precise (that was the approach taken until this PR), but this PR was about making it less precise for future maintenance. The original PR was too general and I provided rules to make it less precise, but not too much so.

I'm fine with either approach.